hera_cal/abscal_funcs.py

```diff
@@ -434,10 +439,11 @@ def delay_lincal(model, data, wgts=None, refant=None, df=9.765625e4, solve_offse
         wgts[k][inf_select] = 0.0
 
         # get delays
-        dly, offset = fft_dly(ratio, wgts=wgts[k], df=df, medfilt=medfilt, kernel=kernel, time_ax=time_ax, freq_ax=freq_ax, solve_phase=solve_offsets)
+        dly, offset = fft_dly(ratio, wgts=wgts[k], df=df, medfilt=medfilt, kernel=kernel, time_ax=time_ax, 
+                              freq_ax=freq_ax, solve_phase=solve_offsets, window=window, edge_cut=edge_cut)
 
         # set nans to zero
-        rwgts = np.ones_like(dly, np.float)
+        rwgts = np.nanmean(wgts[k], axis=freq_ax, keepdims=True)
```

Comment:
Strictly speaking, the mean of the weights isn't quite the optimal weight for the delay, but in practice it's probably fine. I'm not sure off the top of my head what would be optimal, but we can think about it some more if you care.

Reply:
ok yeah we can discuss later